modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java

```diff
@@ -321,6 +337,61 @@ public void testFreeMarkerNotificationPicklistObjectFieldTerm()
 			Collections.singletonList(listEntry.getName()), StringPool.COMMA);
 	}
 
+	@Test
+	public void testFreeMarkerNotificationWithCommerceOrder() throws Exception {
+		CommerceCurrency commerceCurrency =
+			CommerceCurrencyTestUtil.addCommerceCurrency(
+				TestPropsValues.getCompanyId());
+
+		CommerceChannel commerceChannel = CommerceTestUtil.addCommerceChannel(
+			TestPropsValues.getGroupId(), commerceCurrency.getCode());
+
+		CommerceOrder commerceOrder = CommerceTestUtil.addB2CCommerceOrder(
+			TestPropsValues.getUserId(), commerceChannel.getGroupId(),
+			commerceCurrency);
+
+		commerceOrder = CommerceTestUtil.addCheckoutDetailsToCommerceOrder(
+			commerceOrder, TestPropsValues.getUserId(), true, true);
+
+		ObjectDefinition commerceOrderObjectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinitionByClassName(
+				TestPropsValues.getCompanyId(), CommerceOrder.class.getName());
+
+		Map<String, Object> termValuesMap = _createFreeMarkerTermValuesMap(
+			commerceOrderObjectDefinition, commerceOrder,
+			_getTermNames(
+				commerceOrderObjectDefinition.getObjectDefinitionId(),
+				SetUtil.fromArray(
+					"Basic Information", "Workflow Status Information")));
+
+		String body =
+			StringUtil.merge(termValuesMap.keySet(), StringPool.POUND) +
+				StringPool.POUND;
```

Comment:
Hey @pedro-oliveira446 why do we need to contact this second pound?

if we don't need this second pound just inline this `StringUtil.merge(termValuesMap.keySet(), StringPool.POUND)` at:

````java
ObjectAction objectAction = _addNotificationTemplateObjectAction(
			body, DestinationNames.COMMERCE_PAYMENT_STATUS,
			commerceOrderObjectDefinition);
````